torch/csrc/jit/pybind_utils.h

```diff
@@ -74,6 +76,20 @@ inline void findErrorInKwargs(const FunctionSchema& schema, py::kwargs kwargs) {
     }
   }
 }
+
+inline void checkTupleNames(const py::tuple& tuple, const std::vector<std::string> &names) {
+  auto type = tuple.attr("__class__");
+  if (!py::hasattr(type, "_fields"))
+    throw py::cast_error();
+  auto fields = type.attr("_fields").cast<py::tuple>();
```

Comment:
Can't you do something like this?
```cpp
return py::cast<std::vector<std::string>>(type.attr("_fields")) == names;
```

Reply:
@apaszke Yes, you are right. I always forget that `std::vector` ever defined `operator==`....